docker-for-windows/wsl-tech-preview.md

```diff
@@ -1,55 +1,92 @@
+
+4. Select the **Use WSL 2 based engine** check box.
 
-4. Select the **Enable the experimental WSL 2 based engine** check box.
+    If you have installed Docker Desktop on a system that supports WSL 2, this option will be enabled by default.
 5. Click **Apply & Restart**.
 6. Ensure the distribution runs in WSL 2 mode. WSL can run distributions in both v1 or v2 mode.
 
-    To check the WSL mode, run:
+    To check the WSL mode, run
+
+     `wsl -l -v`
+
+    To upgrade your existing Linux distro to v2, run:
+
+    `wsl --set-version (distro name) 2`
+
+    To set v2 as the default version for future installations, run:
+
+    `wsl --set-default-version 2`
+
+7. When Docker Desktop restarts, go to **Settings** > **Resources** > **WSL Integration**.
+
+    WSL Integration will be enabled on your default WSL distribution. To change your default WSL distro, run `wsl --set-default <distro name>`.
+
+    For example, to set Ubuntu as your default WSL distro, run `wsl --set-default ubuntu`.
+
+    Optionally, select any additional distributions you would like to enable WSL 2 on.
+
+    ![WSL 2 Choose Linux distro](images/wsl2-choose-distro.png)
+
+8. Click **Apply & Restart**.
+
+## Develop with Docker and WSL 2
+
+The following section describes how to start developing your applications using Docker and WSL 2. We recommend that you have your code in your default Linux distribution for the best development experience using Docker and WSL 2. After you have enabled WSL 2 on Docker Desktop, you can start working with your code inside the Linux distro and ideally with your IDE still in Windows. This workflow can be pretty straightforward if you are using [VSCode](https://code.visualstudio.com/download).
+
+1. Open VSCode and install the [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) extension. This extension allows you to work with a remote server in the Linux distro and your IDE client still on Windows.
+2. Now, you can start working in VSCode remotely. To do this, open your terminal and type:
+
+    `Wsl`
+
+    `code .`
+
+    This opens a new VSCode connected remotely to your default Linux distro which you can check in the bottom corner of the screen.
 
-    `wsl -l -v`
+    Alternatively, you can type the name of your default Linux distro in your Start menu, open it, and then run `code` .
+3. When you are in VSCode, you can use the terminal in VSCode to pull your code and start working natively from your Windows machine.
 
-    To upgrade to v2, run:
+## Best practices
 
-    `wsl --set-version <distro name> 2`
-7. When Docker Desktop restarts, go to **Settings** > **Resources** > **WSL Integration** and then select from which WSL 2 distributions you would like to access Docker.
+- To get the best out of the file system performance, we recommend that you avoid mounting from the Windows file system (even on a WSL distro). For example, avoid `docker run -v /mnt/c/users:/users`.
```

Comment:
I want this to be followed by "Instead, mount [something else]".

Reply:
@simonferquel Could you provide some guidance here? Thanks.

Reply:
"Instead prefer mounting files living in your distro's local filesystem such as in: `docker run -v ~/my-project:/sources <my-image>`"